src/math/exp_fcn.c

```diff
@@ -0,0 +1,212 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+
+/* Include Files */
+#include <sof/math/exp_fcn.h>
+#include <sof/math/numbers.h>
+#include <sof/common.h>
+#include <rtos/bit.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#define BIT_MASK_Q62P2 0x4000000000000000
+#define CONVERG_ERROR 28823037607936LL	// error smaller than 1e-4,1/2 ^ -44.7122876200884
+#define BIT_MASK_LOW_Q27P5 0x8000000
+#define MIN_IN_VAL -1342177280
+#define MAX_OUT_VAL 1244979733
```

Comment:
The "magic" values seem very specific to `exp_int32()` and used only once there but they're very far away from it and there is nothing in their name that links them to `exp_int32()`.

If they are indeed specific to `exp_int32()` then simply declare them as `const` inside `exp_int32()`; no need to pollute the entire file's namespace with very generic names.

Reply:
I 've added a comment explaining the out of range input value returns clipped output

Reply:
Thanks for adding a comment but my request was not at all about the values, please read it again. It was 100% about WHERE they values are located in the file and HOW they are defined.